src/server/nfc/index.test.ts

```diff
@@ -0,0 +1,108 @@
+import { ObjectId } from 'mongodb';
+import { MongoMemoryServer } from 'mongodb-memory-server';
+import {
+	checkIsAuthorized,
+	fetchUser,
+	query,
+	queryById,
+	toEnum,
+	updateUser,
+} from '../resolvers/helpers';
+import {
+	HackerDbObject,
+	OrganizerDbObject,
+	EventDbObject,
+	ShirtSize,
+	UserDbInterface,
+	UserType,
+} from '../generated/graphql';
+import DB, { Models } from '../models';
+import { isNFCUIDAvailable } from '.';
+
+const hacker: UserDbInterface = {
+	_id: new ObjectId(),
+	createdAt: new Date(),
+	dietaryRestrictions: [],
+	email: 'foo@bar.baz',
+	firstName: 'Foo',
+	lastName: 'Bar',
+	logins: [],
+	preferredName: 'Foo',
+	secondaryIds: [],
+	userType: UserType.Hacker,
+};
+
+const event: EventDbObject = {
+	_id: new ObjectId(),
+	attendees: [],
+	checkins: [],
+	description: 'description',
+	duration: 120,
+	eventType: 'event type',
+	location: 'location',
+	name: 'event',
+	startTimestamp: new Date(),
+	warnRepeatedCheckins: true,
+};
+
+let mongoServer: MongoMemoryServer;
+let dbClient: DB;
+let models: Models;
+
+const testHackerId = new ObjectId();
+const testHacker = ({
+	_id: testHackerId,
+	email: 'foo@bar.com',
+} as unknown) as HackerDbObject;
```

Comment:
unknown type is kinda sketchy

Reply:
It's fine here because this is an incomplete `HackerDbObject` and can't be type casted without unknown. The point is we don't need the other fields and they only clutter the tests

Reply:
what about using Partial<>

Reply:
or https://www.typescriptlang.org/docs/handbook/utility-types.html#picktk